sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -248,30 +269,81 @@ case class Hypot(left: Expression, right: Expression)
 case class Pow(left: Expression, right: Expression)
   extends BinaryMathExpression(math.pow, "POWER") {
   override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
-    defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.pow($c1, $c2)") + s"""
-      if (Double.valueOf(${ev.primitive}).isNaN()) {
-        ${ev.isNull} = true;
-      }
-      """
+    defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.pow($c1, $c2)")
   }
 }
 
 case class Logarithm(left: Expression, right: Expression)
   extends BinaryMathExpression((c1, c2) => math.log(c2) / math.log(c1), "LOG") {
-  def this(child: Expression) = {
-    this(EulerNumber(), child)
-  }
 
-  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
-    val logCode = if (left.isInstanceOf[EulerNumber]) {
-      defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.log($c2)")
+  override def eval(input: InternalRow): Any = {
+    val evalE1 = left.eval(input)
+    if (evalE1 == null || evalE1.asInstanceOf[Double] <= 0.0) {
+      null
     } else {
-      defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.log($c2) / java.lang.Math.log($c1)")
+      val evalE2 = right.eval(input)
+      if (evalE2 == null || evalE2.asInstanceOf[Double] <= 0.0) {
+        null
+      } else {
+        math.log(evalE2.asInstanceOf[Double]) / math.log(evalE1.asInstanceOf[Double])
+      }
     }
-    logCode + s"""
-      if (Double.valueOf(${ev.primitive}).isNaN()) {
+  }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val eval1 = left.gen(ctx)
+    val eval2 = right.gen(ctx)
+    s"""
+      ${eval1.code}
+      boolean ${ev.isNull} = ${eval1.isNull} || ${eval1.primitive} <= 0.0;
+      ${ctx.javaType(dataType)} ${ev.primitive} = ${ctx.defaultValue(dataType)};
+      if (${ev.isNull}) {
         ${ev.isNull} = true;
+      } else {
+        ${eval2.code}
+        if (${eval2.isNull} || ${eval2.primitive} <= 0.0) {
+          ${ev.isNull} = true;
+        } else {
+          ${ev.primitive} = java.lang.Math.${funcName}(${eval2.primitive}) /
+           java.lang.Math.${funcName}(${eval1.primitive});
+        }
       }
     """
   }
+
```

Comment:
Hive's UDFLog doesn't support base in range (0.0, 1.0], it would just eval to null in this case, just act as the commented out code below.

I'm not sure it we also want to behave  like this.